doc/boxes.el

```diff
@@ -4,6 +4,8 @@
 
 ;; Author: Jason L. Shiffer <jshiffer@zerotao.com>
 ;; Maintainer: Jason L. Shiffer <jshiffer@zerotao.com>
+;; Version: 0.0
```

Comment:
Agreed and using 0.0 here means the Emacs package build process will pull the version tag from the Boxes repo, ie use v2.2.0.

I can see an argument for explicitly setting the version in Boxes.el, but that does mean we’ll have to remember to bump the version number here as well when changes are made.

What do you think?

Reply:
Alright, that's fine then!  
I just wanted to avoid that users would need to install "boxes v0.0" in order to get "boxes v2.2.0".

Reply:
Ah, so the `boxes.el` package being installed does not contain boxes, but instead tells the system's package manager to install the most recent boxes.

In this case, I'd say we still couple the two. A `boxes.el` version of 2.2.0 would mean that at the time it was packaged/released, the most recent version of boxes was 2.2.0. In practice, a user might still receive boxes version 1.3 (maybe because he's on Debian bullseye without experimental packages). Therefore, we put a warning into the docstring saying something like: `The concrete version of boxes that this package installs depends on your system's package manager. It might be newer or older than the displayed package version.`

What do you think?

Reply:
In fact I just now realized that the `boxes.el` package installs the boxes binary as well. That's cool, of course! But what do you do if the system doesn't have boxes? There are still some systems around that have Emacs but don't have boxes. (Even though that's an obvious sign they need to straighten out their priorities.)

Reply:
Further explanation of how the Emacs `boxes` package relates to the binary `boxes` package.

- The Emacs package only contains `boxes.el` (& some meta-data).
- By default, whatever version of the `boxes` binary they have installed will be used. If they don't have it installed, an error about missing program `boxes` will be shown.
- If the user uses the `use-package` lisp form I mention in the documentation, then Emacs will check if `boxes` is available on the PATH (which could be any version of `boxes`) and if not call the system package manager to install it. eg on my system it runs `brew install boxes`.

Reply:
Also, there is a dependency between `boxes.el` and `boxes` version, because the `boxes` binary has to support running
```
boxes -q (all)
```
So maybe we should add a note to the package documentation / docstring to say that the `boxes` minimum version is whichever version this was added?

I could even make `boxes.el` check for that minimum version?

Reply:
Yes, sure! I am always in favor of making things better. 😄

Reply:
Hi Thomas,

The Emacs package is now live 😊

https://melpa.org/#/boxes

Looks like 5 people have installed it already!

A slight "fly in the ointment" is the stable package (https://stable.melpa.org/#/boxes) has been built using the `boxes.el` from when the 2.2.0 tag was applied rather than the current version (no real surprise I guess). However no one really uses the stable packages, so we could just ignore that for now and it will be corrected the next time you tag a new release or you could just add a 2.2.1 tag?

Also what's the process for updating the "Emacs Integration" page, as it's obviously way out of date now?

Thanks,
Mike

Reply:
Nice, that's great news! 🎉

Publishing a boxes release is unfortunately still a lot of manual work, so I'd much prefer waiting until enough content for a release has accumulated.

I think we can switch the [Emacs integration](https://boxes.thomasjensen.com/editors-emacs.html) page completely to your package now. What do you think?  
The web site lives on the [gh-pages](https://github.com/ascii-boxes/boxes/tree/gh-pages) branch, where you can modify it via PRs just like the code. It's all markdown, so you don't need HTML or CSS skills to edit the page. In case you want to run the website on your local machine for testing, [here's how](https://boxes.thomasjensen.com/2021/05/website-updates.html#website-on-local-machine).

Once that's done, we should also publish a little news item on the [blog](https://boxes.thomasjensen.com/blog/) to tell people about this!

Let me know if you need support for any of this. And thanks again for this really cool contribution!

Reply:
Understood - I don't think we need to worry about the stable package too much so it will be fine to leave it as-is.

Thanks for your clear instructions - I have pushed my changes to the Integration page now 😊

Thanks,
Mike